File: Semester3/MAP/Seminare/Sem10/Sem10/Decorator/PrinterTaskRunner.cs
namespace Sem10;

public class PrinterTaskRunner : AbstractTaskRunner
{
    public PrinterTaskRunner(TaskRunner runner) : base(runner){ }

    public override void executeOneTask()
    { 
        base.executeOneTask();
        Console.WriteLine("Task executed at : " + DateTime.Now);
    }
    
}

File: Semester3/MAP/laboratoare/lab2/TaskManager/src/Decorator/PrinterTaskRunner.java
package Decorator;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class PrinterTaskRunner extends AbstractTaskRunner {
    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("HH:mm");
    public PrinterTaskRunner(TaskRunner taskRunner) {
        super(taskRunner);
    }

    @Override
    public void executeOneTask(){
        super.executeOneTask();
        System.out.println("Task executed at: " + LocalDateTime.now().format(formatter));
    }
}
